--- a/grimblast	2026-08-28 15:29:31.245301725 +0200
+++ b/grimblast	2026-08-28 15:23:49.734988017 +0200
@@ -153,6 +153,9 @@
 # The actual grim command used is printed to stderr
 screenshot() {
   local file="$1" output="$2"
+  if [[ -n "$file" ]]; then
+    mkdir -p "$(dirname "$file")"
+  fi
   xargs --verbose grim <<<"${CURSOR:+-c} ${SCALE:+-s \"$SCALE\"} -t \"$FILETYPE\" ${output:+-o \"$output\"} ${GEOM:+-g \"$GEOM\"} ${WINDOW:+-T \"$WINDOW\"} \"$file\""
 }
 
@@ -292,6 +295,7 @@
     screenshot - "$OUTPUT" | tee >(wl-copy --type "$(get-mime-type)") || die "Clipboard error"
     notify::ok "$WHAT copied to buffer and piped to stdout"
   else
+    mkdir -p "$(dirname "$file")"
     screenshot - "$OUTPUT" | tee "$file" | wl-copy --type "$(get-mime-type)" || die "Clipboard error"
     title="Screenshot of $WHAT"
     message="$WHAT copied to buffer and saved to $file"
